### ISSUE TITLE
comment --pr=true default

### ISSUE THREAD

**Issue by casperdcl** [iterative/cml#1139]
https://github.blog/changelog/2022-08-04-commit-comments-no-longer-appear-in-the-pull-request-timeline/ meaning the `--pr` flag is now mandatory for CML reports to show up in PRs. Kinda a UX breaking change potentially.

Options (pick just one):

1. update docs & examples, adding `--pr` everywhere (breaking change, https://github.com/iterative/cml.dev/issues/331)

<details><summary>also auto-detect this/print warnings?</summary>

```bash
$ cml comment ...
(ok)
$ cml pr ...
(ok)
$ cml comment ...
WARNING: did you forget `--pr` flag?
```

</details>

2. `cml comment ... [--pr=true]` by default (UX friendly)
   + iff not in a PR, print a warning and fallback to `--pr=false` (commit-comment)
   + users need explicit `--pr=false`/`--no-pr` to use commit-comments in PRs (edge case)
3. `cml comment ... [--pr=auto]` by default (better fail-not-warn error handling)
   + try to PR-comment, fallback to commit-comment
   + explicit `--pr={true,false}` throws error on failure (no fallback)
   + however, is this implementation possible? does yargs allow `--flag=foo` == `"foo"` && `--flag` == `true` (latter needed for backwards compat)?

**Comment by 0x2b3bfa0** [comment]
### Historical Context

Traditionally, `cml send-comment` has operated on individual commits rather than pull requests.

We promoted a workflow where users opened a single pull request per “experimentation session” and then pushed [potentially] several commits, each one with a different “experiment” like modifying an hyperparameter. Reports for every of these changes would be available as a commit comment as soon as the training process finished.

Then, there was a request for creating pull request comments rather than commit comments (#735) to avoid losing comments when rebasing a pull request.

**Comment by 0x2b3bfa0** [comment]
### Suggested solution

Commit comments are unequivocally tied to a commit, but aren't supported on every forge.[^1] Wouldn't it be better if we used `--pr` everywhere and, perhaps, added a footer showing the commit hash?

[^1]: Bitbucket, as usual, does not support commit comments.

**Comment by tasdomas** [comment]
One problem with requiring 
```
      - uses: actions/checkout@v3
        with:
          ref: ${{ github.event.pull_request.head.sha }}
```
in the workflow definition for `cml comment create --pr` to work is that we're not including any master changes in the context of the action.

I think we should augment cml drivers with forge-specific logic for determining the pull request (if there is one) to add comments to.

**Comment by dacbd** [comment]
> we're not including any master changes in the context of the action.

I'm not following, can you elaborate?

**Comment by casperdcl** [comment]
I think @tasdomas means checking `$GITHUB_HEAD_REF` instead of requiring explicit `checkout.with.ref: ${{ github.event.pull_request.head.sha }}`

**Comment by 0x2b3bfa0** [comment]
> _What about [this](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-commits/#api-repositories-workspace-repo-slug-commit-commit-comments-post)?_

Probably a layer-8 error on my side 🤔

**Comment by casperdcl** [comment]
opened #1223

**Comment by dacbd** [comment]
@casperdcl, with the improved options and logic with the introduction of `cml comment create --target=x`, I think we can close this?

**Comment by casperdcl** [comment]
Yes closed by #1228
